kvserver: `TestPromoteNonVoterInAddVoter` flaked on `release-23.1`

Seen in a [CI failure](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/10734994?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true).

```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestPromoteNonVoterInAddVoter2438547430
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2248: condition failed to evaluate within 3m45s: Range counts are not all equal
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestPromoteNonVoterInAddVoter2438547430
--- FAIL: TestPromoteNonVoterInAddVoter (462.14s)
```

Jira issue: CRDB-29255

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2211,10 +2211,11 @@ func TestPromoteNonVoterInAddVoter(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// This test is slow under stress and can time out when upreplicating /
+	// This test is slow under stress/race and can time out when upreplicating /
 	// rebalancing to ensure all stores have the same range count initially, due
 	// to slow heartbeats.
 	skip.UnderStress(t)
+	skip.UnderRace(t)
 
 	ctx := context.Background()
 

